lib/yard/handlers/ruby/method_handler.rb

```diff
@@ -19,10 +19,10 @@ class YARD::Handlers::Ruby::MethodHandler < YARD::Handlers::Ruby::Base
 
     nobj = P(namespace, nobj.value) while nobj.type == :constant
     obj = register MethodObject.new(nobj, meth, mscope) do |o|
-      o.signature = method_signature
       o.explicit = true
       o.parameters = args
     end
+    obj.signature = method_signature
```

Comment:
It seems like this should be fixed in `MethodObject.new` directly-- I think it's reasonable to expect that setting `o.signature` inside the yielded block would override the default signature, so this is probably an ordering bug inside of #initialize rather than here.